File: collatzconjecture/kotlin/src/main/kotlin/collatzconjecture/CollatzConjecture.kt
package collatzconjecture

import java.lang.IllegalArgumentException

fun calculateSteps(number: Int): Int {

    if (number <= 0) throw IllegalArgumentException()

    var n = number
    var steps = 0

    while (n != 1) {
        n = if (n.rem(2) == 0) n / 2 else n * 3 + 1
        steps++
    }

    return steps
}

File: collatzconjecture/javascript/js/collatzconjecture.js
export function calculateSteps(number) {
    if (!Number.isInteger(number) || number <= 0) {
        throw new Error("Invalid argument")
    }

    let n = number;
    let steps = 0;

    while (n != 1) {
        n = n % 2 === 0 ? n / 2 : n * 3 + 1;
        steps++;
    }

    return steps;
}